src/Featurizers/DateTimeFeaturizer.cpp

```diff
@@ -371,19 +379,19 @@ void DateTimeTransformer::save(Archive & ar) const /*override*/ {
 // ----------------------------------------------------------------------
 // ----------------------------------------------------------------------
 void DateTimeTransformer::execute_impl(InputType const &input, CallbackFunction const &callback) /*override*/ {
-    std::chrono::time_point<std::chrono::system_clock> time(std::chrono::seconds {input} );
-    TimePoint result = TimePoint(time);
+    static constexpr long long const        secondsPerDay(60 * 60 * 24);
+
+    TimePoint                               result(input);
 
     if (!_dateHolidayMap.empty()) {
-        //Normalize dateKey by cast one day range of time into an exact time
-        //86400 is the total number of seconds in a day, and as the holiday time is provided in a manner(00:00:00 in one
-        //day, so we need to consider any time that falls into this day.
-        InputType dateKey = (input - input % 86400);
-
-        auto x = _dateHolidayMap.find(dateKey);
-        if (x != _dateHolidayMap.end()) {
-            result.holidayName = x->second;
-        }
+        // Values in the holiday map are based on midnight of the corresponding day. When looking up holidays for this time,
+        // remove any seconds since midnight.
+        long long const                     totalSeconds(std::chrono::time_point_cast<std::chrono::seconds>(input).time_since_epoch().count());
+        long long const                     holidayKey(totalSeconds - (totalSeconds % secondsPerDay));
+        HolidayMap::const_iterator const    iter(_dateHolidayMap.find(static_cast<std::int64_t>(holidayKey)));

```

Comment:
>static_cast<std::int64_t> [](start = 70, length = 25)

I'd move that to totalSeconds and convert everything to int64_t?

Reply:
I'm not sure I completely follow your comment, but I think that you are saying that either the map key should be long long or the holidaySeconds should be int64?